### PR TITLE
ARROW-13760: [C++] Bump required Protobuf when using Flight

### PR DESCRIPTION
**Pull request by lidavidm**
This is to support the FlightSQL proposal in #10906.

### CODE REVIEW

**Comment by lidavidm** [comment]
@github-actions crossbow submit -g nightly

**Comment by lidavidm** [comment]
Things look to generally be passing here, minus some flakes and some failures from master.

File: cpp/cmake_modules/ThirdpartyToolchain.cmake [hunk 1]
```diff
@@ -1371,19 +1371,32 @@ macro(build_protobuf)
...
+        "-DCMAKE_CXX_FLAGS=${PROTOBUF_CXX_FLAGS}"
+        "-DCMAKE_C_FLAGS_${UPPERCASE_BUILD_TYPE}=${PROTOBUF_C_FLAGS}"
+        "-DCMAKE_CXX_FLAGS_${UPPERCASE_BUILD_TYPE}=${PROTOBUF_CXX_FLAGS}")
     if(MSVC AND NOT ARROW_USE_STATIC_CRT)
       list(APPEND PROTOBUF_CMAKE_ARGS "-Dprotobuf_MSVC_STATIC_RUNTIME=OFF")
     endif()
     if(ZLIB_ROOT)
       list(APPEND PROTOBUF_CMAKE_ARGS "-DZLIB_ROOT=${ZLIB_ROOT}")
     endif()
+    list(APPEND PROTOBUF_CMAKE_ARGS "-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON")
```
Comment (kou) [inline cpp/cmake_modules/ThirdpartyToolchain.cmake:1399]: Is this just for debug?

**Review by kou — APPROVED** [review]
+1

File: cpp/cmake_modules/ThirdpartyToolchain.cmake (hunk 1 above)
Comment (lidavidm) [reply cpp/cmake_modules/ThirdpartyToolchain.cmake]: Ah, yeah, sorry. I've taken this out.

**Comment by lidavidm** [comment]
Thanks for checking this over!

**Pull request closed without merging**